controllers/datadogagent/feature/dogstatsd/feature.go

```diff
@@ -140,20 +140,18 @@ func (f *dogstatsdFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) (reqComp feat
 // ManageDependencies allows a feature to manage its dependencies.
 // Feature's dependencies should be added in the store.
 func (f *dogstatsdFeature) ManageDependencies(managers feature.ResourceManagers, components feature.RequiredComponents) error {
-	if f.hostPortEnabled {
-		// agent local service
-		if component.ShouldCreateAgentLocalService(managers.Store().GetVersionInfo(), f.forceEnableLocalService) {
-			apmPort := []corev1.ServicePort{
-				{
-					Protocol:   corev1.ProtocolUDP,
-					TargetPort: intstr.FromInt(int(f.hostPortHostPort)),
-					Port:       f.hostPortHostPort,
-					Name:       apicommon.DogstatsdHostPortName,
-				},
-			}
-			if err := managers.ServiceManager().AddService(f.localServiceName, f.owner.GetNamespace(), nil, apmPort, nil); err != nil {
-				return err
-			}
+	// agent local service
+	if component.ShouldCreateAgentLocalService(managers.Store().GetVersionInfo(), f.forceEnableLocalService) {
+		dsdPort := []corev1.ServicePort{
+			{
+				Protocol:   corev1.ProtocolUDP,
+				TargetPort: intstr.FromInt(apicommon.DefaultDogstatsdPort),
+				Port:       apicommon.DefaultDogstatsdPort,
```

Comment:
think we need to add another `if` condition to override the `apicommon.DefaultDogstatsdPort` by the `f.hostPortHostPort`. 
the usecase that I see is when the agent pod is configured to run with `host network`, the container port is corresponding to the host port. 

to know if hostNetwork is enable we have this function https://github.com/DataDog/datadog-operator/blob/6beacdb5898bf470669c846526c8ebe1dcdeebea/apis/datadoghq/v2alpha1/utils.go#L73

and then it is used here https://github.com/DataDog/datadog-operator/blob/6beacdb5898bf470669c846526c8ebe1dcdeebea/controllers/datadogagent/feature/dogstatsd/feature.go#L197-L199